src/components/Image/Image.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback, useState } from 'react';
+import { Image as RNImage, StyleSheet, View, ViewStyle } from 'react-native';
+
+import Button from '../Button';
+import styles, { DEFAULT_IMAGE_SIZE } from './image.style';
+import ImageShimmer from './ImageShimmer';
+
+interface Props {
+  source: number | { uri: string };
+  style?: ViewStyle;
+  isLoading: boolean;
+}
+
+const Image = ({ source, style, isLoading }: Props) => {
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const [hasError, setHasError] = useState<boolean>(false);
+
+  const flattenedStyles = StyleSheet.flatten(style);
+  const width = (flattenedStyles?.width as number) || DEFAULT_IMAGE_SIZE;
+  const height = (flattenedStyles?.width as number) || DEFAULT_IMAGE_SIZE;
+
+  const onError = useCallback(() => {
+    setHasError(true);
+  }, []);
+
+  const showShimmer = (!isLoaded && !hasError) || isLoading;
+
+  return (
+    <View style={[styles.imageContainer, style]}>
+      <RNImage
+        style={{ width, height }}
+        source={source}
+        onError={onError}
+        onLoad={() => setIsLoaded(true)}
+      />
+      {showShimmer && <ImageShimmer width={width} height={height} />}
+      {hasError && <Button style={styles.retryButton} title={'Error: Tap to retry'} />}
```

Comment:
Need to make this flexible for other projects. 

So ideally we need to be able to pass our own **errorTitle / errorTitleStyle**. Also might be good to pass a **renderErrorContent** inside the button just incase we want to render a icon or other component inside.  And of course we need a callback so we can do something with the retry option.